.pre-commit-config.yaml

```diff
@@ -0,0 +1,52 @@
+# If you want use pre-commit, you need to install pre-commit package.
+# $ pip install pre-commit
+# $ pre-commit install
+
+default_language_version:
+  python: python3
+
+repos:
+  - repo: https://github.com/psf/black
+    rev: 24.4.0
+    hooks:
+      - id: black
+  - repo: https://github.com/pycqa/flake8
+    rev: 7.0.0
+    hooks:
+      - id: flake8
+        exclude: .venv|venv|build|tutorial|.asv
+        args: [
+            "--max-line-length=99",
+            "--ignore=E203,E704,W503",
+            "--statistics",
+        ]
+  - repo: https://github.com/pycqa/isort
+    rev: 5.13.2
+    hooks:
+      - id: isort
+  - repo: https://github.com/pre-commit/mirrors-mypy
+    rev: v1.9.0
+    hooks:
+    - id: mypy
+      additional_dependencies: [
+        "types-PyYAML",
+        "types-redis",
+        "types-setuptools",
+        "types-tqdm",
+        "typing_extensions>=3.10.0.0",
+      ]
+      exclude: .venv|venv|build|docs|tutorial|optuna/storages/_rdb/alembic
+      args: [
+        --warn-unused-configs,
+        --disallow-untyped-calls,
+        --disallow-untyped-defs,
+        --disallow-incomplete-defs,
+        --check-untyped-defs,
+        --no-implicit-optional,
+        --warn-redundant-casts,
+        --strict-equality,
+        --extra-checks,
+        --no-implicit-reexport,
+        --ignore-missing-imports,
+        --allow-redefinition,
```

Comment:
Seems this line does not exist in [setup.cfg](https://github.com/optuna/optuna/blob/master/setup.cfg).